### PR TITLE
Feature/release v0.3.3

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR prepares develop for 0.3.3 release.

## Fixes

-

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

-

### CODE REVIEW

File: HISTORY.md [hunk 1]
```diff
@@ -1,5 +1,23 @@
 # Release History
 
+## 0.3.3 (2020-05-22)
+
+- Adds option to pass ledger apis to aea builder
+- Refactors decision maker refactor: separates interface and implementation; adds loading mechanisms so framework users can provide their own implementation
```
Comment (marcofavorito) [inline HISTORY.md:6]: `Refactors decision maker refactor` typo?

File: docs/version.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-The current version of the Autonomous Economic Agent framework is `0.3.2`. The framework is under rapid development with frequent breaking changes.
+The current version of the Autonomous Economic Agent framework is `0.3.3`. The framework is under rapid development with frequent breaking changes.
```
Comment (marcofavorito) [inline docs/version.md:1]: idea: replace with 
PyPI

File: HISTORY.md [hunk 3]
```diff
@@ -1,5 +1,23 @@
...
+- Adds async and sync agent loop implementations; agent can be run in both sync and async mode
+- Completes transition to atomic cli commands (fetch, generate, scaffold)
+- Refactors dialogue api: adds much simplified api; updates generator accordingly; updates skills
+- Adds support for crypto module extensions: framework users can register their own crypto module
+- Adds crypto module and ledger support for cosmos
+- Adds simple-oef (soef) connection
+- Adds p2p_libp2p connection for true p2p connectivity via the internet
```
Comment (marcofavorito) [inline HISTORY.md:13]: "via the internet" maybe we can omit it.

File: aea/cli/interact.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    StubConnection,
+)
+from aea.mail.base import Envelope, InBox, Multiplexer, OutBox
+
+
+class InterruptInputException(Exception):
+    """An exception to mark an interuption event."""
+
+
+def try_construct_envelope() -> Optional[Envelope]:
```
Comment (DavidMinarsch) [inline aea/cli/interact.py:40]: First iteration of a command to interact with a running aea via the stub connection. Helps during development...

File: aea/helpers/dialogue/base.py [hunk 5]
```diff
@@ -581,7 +571,7 @@ def update(self, message: Message) -> Optional[Dialogue]:
             dialogue = self._create_opponent_initiated(
                 dialogue_opponent_addr=message.counterparty,
                 dialogue_reference=dialogue_reference,
-                role=Dialogue.role_from_first_message(message),
```
Comment (DavidMinarsch) [inline aea/helpers/dialogue/base.py:584]: That was wrongly implemented, the concrete/derived class is needed..Fixing this had a lot of downstream effects

File: packages/fetchai/skills/carpark_client/dialogues.py [hunk 6]
```diff
@@ -59,16 +59,6 @@ def __init__(
         )
         self.proposal = None  # type: Optional[Description]
 
-    @staticmethod
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/carpark_client/dialogues.py:62]: Moving this to dialogues means we only have one ABC

File: packages/fetchai/skills/carpark_client/dialogues.py [hunk 7]
```diff
@@ -82,22 +72,28 @@ def __init__(self, **kwargs) -> None:
...
+    def role_from_first_message(message: Message) -> Dialogue.Role:
+        """
+        Infer the role of the agent from an incoming or outgoing first message
+
+        :param message: an incoming/outgoing first message
+        :return: the agent's role
+        """
+        return FipaDialogue.AgentRole.BUYER
+
+    def create_dialogue(
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/carpark_client/dialogues.py:85]: Reimplementation as we need the derived class `Dialogue` which has been extended above...

**Pull request merged**